src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
+// import { thisTypeAnnotation } from "@babel/types";
+
+type Shape = 'triangle'|'circle'|'rectangle';
+type Color = 'red'|'green'|'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea():number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
 
-}
+  constructor(
+    public color: Color,
+    public a:number,
+    public b:number,
+    public c:number,
+  ) {
+    this.shape = 'triangle';
+
+    if (this.a <= 0
+      || this.b <= 0
+      || this.c <= 0
+      || this.a + this.b <= this.c
+      || this.a + this.c <= this.b
+      || this.b + this.c <= this.a) {
+      throw new Error('you can not create triangle with this sides');
```

Comment:
Would be good to explain what exactly was wrong. Why user's values are incorrect and which are correct
```suggestion
    if (
      this.a <= 0
      || this.b <= 0
      || this.c <= 0
      || this.a + this.b <= this.c
      || this.a + this.c <= this.b
      || this.b + this.c <= this.a
    ) {
      throw new Error('you can not create triangle with this sides');
```